group-coordinator/src/main/java/org/apache/kafka/coordinator/group/OffsetMetadataManager.java

```diff
@@ -333,6 +348,76 @@ public CoordinatorResult<OffsetCommitResponseData, Record> commitOffset(
         return new CoordinatorResult<>(records, response);
     }
 
+    /**
+     * Handles an OffsetDelete request.
+     *
+     * @param request The OffsetDelete request.
+     *
+     * @return A Result containing the OffsetDeleteResponseData response and
+     *         a list of records to update the state machine.
+     */
+    public CoordinatorResult<OffsetDeleteResponseData, Record> deleteOffsets(
+        OffsetDeleteRequestData request
+    ) throws ApiException {
+        final Group group = validateOffsetDelete(request);
+        final List<Record> records = new ArrayList<>();
+        final OffsetDeleteResponseData.OffsetDeleteResponseTopicCollection responseTopicCollection =
+            new OffsetDeleteResponseData.OffsetDeleteResponseTopicCollection();
+        final OffsetDeleteResponseData response = new OffsetDeleteResponseData();
+        final TimelineHashMap<String, TimelineHashMap<Integer, OffsetAndMetadata>> offsetsByTopic =
+            offsetsByGroup.get(request.groupId());
+
+        request.topics().forEach(topic -> {
+            final OffsetDeleteResponseData.OffsetDeleteResponsePartitionCollection responsePartitionCollection =
+                new OffsetDeleteResponseData.OffsetDeleteResponsePartitionCollection();
+            final boolean subscribedToTopic = group.isSubscribedToTopic(topic.name());
+            final TimelineHashMap<Integer, OffsetAndMetadata> offsetsByPartition = offsetsByTopic.get(topic);
+
+            topic.partitions().forEach(partition -> {
+                final OffsetDeleteResponseData.OffsetDeleteResponsePartition responsePartition =
+                    new OffsetDeleteResponseData.OffsetDeleteResponsePartition().setPartitionIndex(partition.partitionIndex());
+                if (subscribedToTopic) {
+                    responsePartition.setErrorCode(Errors.GROUP_SUBSCRIBED_TO_TOPIC.code());
+                } else if (offsetsByPartition != null && offsetsByPartition.containsKey(partition)) {
+                    records.add(RecordHelpers.newOffsetCommitTombstoneRecord(
+                        request.groupId(),
+                        topic.name(),
+                        partition.partitionIndex()
+                    ));
+                }
+                responsePartitionCollection.add(responsePartition);
+            });
+
+            final OffsetDeleteResponseData.OffsetDeleteResponseTopic responseTopic =
+                new OffsetDeleteResponseData.OffsetDeleteResponseTopic().setPartitions(responsePartitionCollection);
+            responseTopicCollection.add(responseTopic);
+        });
+        response.setTopics(responseTopicCollection);
+
+        return new CoordinatorResult<>(records, response);
+    }
+
+    /**
+     * Handles an GroupDelete request.
```

Comment:
nit: Handles "a"

maybe we can reword this to "Deletes offsets as part of a DeleteGroups request."